### PR TITLE
Addin syntax for related private functions

### PR DESCRIPTION
**Pull request by victorlcampos**
I take this syntax from progdav book, position 2095(mobi)

### CODE REVIEW

**Comment by ericmj** [comment]
Convention in elixir core and most libraries is `do_sum`, but I have seen `_sum` in the wild as well although a lot less often.

**Comment by victorlcampos** [comment]
Something like this?

**Comment by ericmj** [comment]
Yes

File: README.md [hunk 1]
```diff
@@ -381,7 +381,15 @@ If you're looking for other projects to contribute to please see the
     # checks if var is cool
   end
   ```
+* Private function with the same name of public functions, should start with `do_`.
 
+  ```elixir
+  def sum(list), do: _sum(list, 0)
+  
+  #private methods
+  def do_sum([], total), do: total
```
Comment (doomspork) [inline README.md]: Shouldn't these be `defp`?

Comment (whatyouhide) [inline README.md]: Also, a space after the `#` in the comment is missing :)

File: README.md [hunk 2]
```diff
@@ -700,4 +708,4 @@ Please Tweet, star, and let any Elixir programmer know about
 [Elixir]: http://elixir-lang.org
 [Hex]: https://hex.pm/packages
 [license]: http://creativecommons.org/licenses/by/3.0/deed.en_US
-[Ruby community style guide]: https://github.com/bbatsov/ruby-style-guide
\ No newline at end of file
+[Ruby community style guide]: https://github.com/bbatsov/ruby-style-guide
```
Comment (doomspork) [inline README.md]: Out of curiosity, what did this change?

Comment (victorlcampos) [inline README.md]: I realy don't know, I edited using Github edit. Sry

File: README.md (hunk 1 above)
Comment (victorlcampos) [inline README.md]: done

**Comment by victorlcampos** [comment]
Hi...
Something like this?

**Comment by whatyouhide** [comment]
@victorlcampos in Elixir there are no "methods", just functions. Please reword the comment to `# private functions` :) thanks!

File: README.md (hunk 2 above)
Comment (Gazler) [inline README.md]: The new line at the end was removed. See: https://github.com/victorlcampos/elixir_style_guide/commit/387ee554a54f20e2c2aff3b276d7607fdcf594ec.patch

```diff
@@ -700,4 +708,4 @@ Please Tweet, star, and let any Elixir programmer know about
 [Elixir]: http://elixir-lang.org
 [Hex]: https://hex.pm/packages
 [license]: http://creativecommons.org/licenses/by/3.0/deed.en_US
-[Ruby community style guide]: https://github.com/bbatsov/ruby-style-guide
\ No newline at end of file
+[Ruby community style guide]: https://github.com/bbatsov/ruby-style-guide
```

**Comment by victorlcampos** [comment]
@whatyouhide done =)

File: README.md [hunk 3]
```diff
@@ -381,7 +381,15 @@ If you're looking for other projects to contribute to please see the
     # checks if var is cool
   end
   ```
+* Private function with the same name of public functions, should start with `do_`.
 
+  ```elixir
+  def sum(list), do: _sum(list, 0)
```
Comment (whatyouhide) [inline README.md]: `do_sum` instead of `_sum`?

File: README.md [hunk 4]
```diff
@@ -381,7 +381,15 @@ If you're looking for other projects to contribute to please see the
...
+  
+  # private functions
+  defp do_sum([], total), do: total
+  defp do_sum([ head | tail ], total), do: _sum(tail, head+total)
```
Comment (whatyouhide) [inline README.md]: `do_sum` instead of `sum`?

Comment (whatyouhide) [inline README.md]: Also, could you use `[head|tail]` (with no spacing) as that's the convention right now (and that's how it's used in the rest of this style guide)?

**Comment by whatyouhide** [comment]
@victorlcampos could you squash all the commits into one so that we don't clutter the history/

**Comment by victorlcampos** [comment]
@whatyouhide, i changed from '_' to 'do_' following @ericmj comment:

"Convention in elixir core and most libraries is do_sum, but I have seen _sum in the wild as well although a lot less often."

Yes, I can squash =)

**Comment by whatyouhide** [comment]
@victorlcampos yes, but you're using both `_sum` as well as `do_sum` here :) I meant to say, use only the `do_` version.

**Comment by victorlcampos** [comment]
Done =)

**Comment by victorlcampos** [comment]
And thx for help me to contribute =)
I'm very happy to help

File: README.md [hunk 5]
```diff
@@ -381,7 +381,15 @@ If you're looking for other projects to contribute to please see the
     # checks if var is cool
   end
   ```
+* Private function with the same name of public functions, should start with `do_`.
```
Comment (whatyouhide) [inline README.md]: I'd remove the comma here and I think we could be good to go after squashing?

File: README.md [hunk 6]
```diff
@@ -381,7 +381,15 @@ If you're looking for other projects to contribute to please see the
...
 
+  ```elixir
+  def sum(list), do: do_sum(list, 0)
+  
+  # private functions
+  defp do_sum([], total), do: total
+  defp do_sum([head|tail], total), do: do_sum(tail, head+total)
```
Comment (whatyouhide) [inline README.md]: Oh, and the space around the `+`: `head + total`

**Comment by victorlcampos** [comment]
Done =)

**Comment by victorlcampos** [comment]
@whatyouhide what do you think? It's ok to go?

**Comment by whatyouhide** [comment]
@victorlcampos looks good :) I'm not actually a maintainer so let's wait for one of them to merge this!

**Comment by Gazler** [comment]
Thanks!

**Pull request merged**